tests/composition/test_composition.py

```diff
@@ -3436,6 +3436,62 @@ def test_call_beforeafter_values_twopass(self):
+                            #  pytest.param('LLVMExec', marks=pytest.mark.llvm),
+                             pytest.param('LLVMRun', marks=pytest.mark.llvm),
+                             pytest.param('PTXExec', marks=[pytest.mark.llvm, pytest.mark.cuda]),
+                             pytest.param('PTXRun', marks=[pytest.mark.llvm, pytest.mark.cuda])
+                             ])
+    @pytest.mark.parametrize(["condition", "expected_result"],[
+                             (pnl.EveryNCalls, [[.25, .25]]),
+                             (pnl.WhenFinished, [[1.0, 1.0]]),
+                             (pnl.All, [[1.0, 1.0]]),
+                             (pnl.AllHaveRun, [[.05, .05]]), #FIXME: Infinite hang
+                             (pnl.Always, [[0.05, 0.05]]),
+                            #  (pnl.AtPass, [[.3, .3]]), #FIXME: Differing result between llvm and python
+                             (pnl.AtTrial,[[0.05, 0.05]]),
+                            #  (pnl.Never), #TODO: Find a good test case for this!
+                            ])
+    def test_scheduler_conditions(self, mode, condition, expected_result):
+        print(mode, condition)
+        decisionMaker = pnl.DDM(
+                        function=pnl.DriftDiffusionIntegrator(starting_point=0,
+                                                              threshold=1,
+                                                              noise=0.0),
+                        reinitialize_when=pnl.AtTrialStart(),
+                        output_ports=[pnl.DECISION_VARIABLE, pnl.RESPONSE_TIME],
+                        name='DDM')
+
+        response = pnl.ProcessingMechanism(size=2, name="GATE")
+
+        comp = pnl.Composition()
+        comp.add_node(decisionMaker)
+        comp.add_node(response)
+        comp.add_projection(pnl.MappingProjection(), sender=decisionMaker, receiver=response)
+
+        if condition is pnl.EveryNCalls:
+            comp.scheduler.add_condition(response, condition(decisionMaker, 5))
+        elif condition is pnl.WhenFinished:
+            comp.scheduler.add_condition(response, condition(decisionMaker))
+        elif condition is pnl.All:
+            comp.scheduler.add_condition(response, condition(pnl.WhenFinished(decisionMaker)))
+        elif condition is pnl.AllHaveRun:
+            comp.scheduler.add_condition(response, condition(decisionMaker))
+        elif condition is pnl.Always:
+            comp.scheduler.add_condition(response, condition())
+        elif condition is pnl.AtPass:
+            comp.scheduler.add_condition(response, condition(5))
+        elif condition is pnl.AtTrial:
+            comp.scheduler.add_condition(response, condition(0))
+        result = comp.run([0.05], bin_execute=mode)
+        result = [x for x in np.array(result).flatten()] #HACK: The result is an object dtype in Python mode for some reason?
```

Comment:
use the same kind of workaround as other DDM tests. it's better to not have multiple versions of the same thing.